coalib/bearlib/aspects/__init__.py

```diff
@@ -0,0 +1,308 @@
+        self.description = cleandoc(description)
+        self.cast_type = cast_type
+        self.example_values = example_values
+        self.default = default
+
+
+class Aspect:
+    '''
+    This is the ``Aspect`` base class. Each aspect may have sub-aspects,
+    which are ``Aspect`` objects themselves. An aspect also contains
+    documentation, instantiated as an ``AspectDocumentation`` object.
+
+    There is a root ``Aspect`` object in which every other aspect lives.
+
+    >>> Root = Aspect()
+
+    This is a placeholder aspect through which every other aspect can be
+    accessed.
+
+    To create new aspect, simply call the ``new_subaspect`` method with
+    the subaspect's name as a param, along with its documentation (mandatory
+    for each aspect):
+
+    >>> Root.new_subaspect(
+    ...     "Redundancy",
+    ...     doc=AspectDocumentation(
+    ...         """
+    ...         This aspect describes redundancy in your source code. Those are
+    ...         usually places where you can strip away source code in order to
+    ...         make your codebase better maintainable but keeping your full
+    ...         functionality.
+    ...         """, '...', '...', '...'
+    ...     )
+    ... )
+
+    This aspect is now accesible at ``Root.Redundancy``. Simply call
+    ``new_subaspect`` to keep creating new sub-aspects:
+
+    >>> Root.Redundancy.new_subaspect(
+    ...     "Clone",
+    ...     doc=AspectDocumentation(
+    ...         "This aspect detects clones in your source code...",
+    ...         '...', '...' ,'...'
+    ...     )
+    ... )
+
+    And this aspect will be available at ``Root.Redundancy.Clone``.
+
+    Trying to access an undefined aspect will raise a ``NameError``:
+
+    >>> Root.UnknownAspect is None
+    Traceback (most recent call last):
+      ...
+    NameError
+
+    And of course, you can create settings for your aspects:
+
+    >>> Root.Redundancy.Clone.settings = [
+    ...     AspectSetting("min_clone_token",
+    ...                   "The number of tokens that have to be "
+    ...                   "equal for it to be detected as a code clone",
+    ...                   int, default=20)
+    ... ]
+
+    ``settings`` can be defined for any aspect by creating a list of
+    ``AspectSetting`` objects with 3 params: the setting name,
+    a documentation string, and a cast type (in this case ``int``
+    will be applied to the given setting). You may also give a
+    fourth optional param - the default value for the setting.
+
+    All bears implementing a given aspect should implement all the settings
+    given and documented by it. Additional bear specific settings might be
+    added to allow further configuration.
+
+    You can also give the settings at object instantiation using the
+    ``settings`` param. Each setting can also contain a param called
+    ``example_values`` - these are helpful for the user to understand
+    what the setting does and what values it may take:
```

Comment:
Actually you can use those for quickstart, maybe rename it to suggested values instead of example?